src/Orleans/Providers/ProviderTypeLoader.cs

```diff
@@ -87,6 +87,14 @@ private void ProcessAssemblyLocally(Assembly assembly)
 
         private static void ProcessNewAssembly(object sender, AssemblyLoadEventArgs args)
         {
+#if !NETSTANDARD
+            // If the assembly is loaded for reflection only avoid processing it.
+            if (args.LoadedAssembly.ReflectionOnly)
```

Comment:
I meant, we would only fully load the assembly once we first did a pass via reflection-only and found that there is at least one type that implements IProvider (unless I'm not remembering correctly). That step is separate from this, or is this the step that actually scans the reflection only assemblies and decides whether to fully load it?